### PR TITLE
Support proxy contract in turing credit contract

### PR DESCRIPTION
**Pull request by boyuan-chen**
* Allow the proxy contract to set the owner

### CODE REVIEW

File: packages/contracts/contracts/L2/predeploys/BobaTuringCredit.sol [hunk 1]
```diff
@@ -84,6 +91,19 @@ contract BobaTuringCredit is Ownable {
...
+   *
+   * @param _newOwner new owner address
+   */
+  function transferOwnership(address _newOwner)
+    public
+    onlyOwner
+  {
+    owner = _newOwner;
+    emit TransferOwnership(msg.sender, _newOwner);
+  }
```
Comment (InoMurko) [inline packages/contracts/contracts/L2/predeploys/BobaTuringCredit.sol]:
```
 /**
   * @dev Allows the current owner to transfer control of the contract to a newOwner.
   * @param newOwner The address to transfer ownership to.
   */
  function transferOwnership(address _newOwner) onlyOwner {
    if (_newOwner != address(0)) {
      owner = _newOwner;
      emit TransferOwnership(msg.sender, _newOwner);
    }
  }
```

Comment (boyuan-chen) [reply packages/contracts/contracts/L2/predeploys/BobaTuringCredit.sol]: Fixed

Comment (CAPtheorem) [reply packages/contracts/contracts/L2/predeploys/BobaTuringCredit.sol]: Public yes/no?

File: packages/contracts/contracts/L2/predeploys/BobaTuringCredit.sol [hunk 2]
```diff
@@ -100,6 +100,7 @@ contract BobaTuringCredit {
     public
     onlyOwner
   {
+    require(_newOwner != address(0));
     owner = _newOwner;
     emit TransferOwnership(msg.sender, _newOwner);
```
Comment (wsdt) [inline packages/contracts/contracts/L2/predeploys/BobaTuringCredit.sol:105]: _msgSender() -> Context.sol probably a good idea?

**Review by CAPtheorem — APPROVED** [review]

**Review by CAPtheorem — APPROVED** [review]

**Pull request merged**